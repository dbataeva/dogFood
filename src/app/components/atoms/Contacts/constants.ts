export const TEXT_MAP = {
	contactUs: 'Мы на связи',
	phone: '8 (999) 00-00-00',
	email: 'dogfood.ru@gmail.com',
};

export const TEST_ID_MAP = {
	phoneAndEmail: 'phoneAndEmail',
	telegram: 'telegram',
	watsApp: 'watsApp',
	instagram: 'instagram',
	facebook: 'facebook',
	twitter: 'twitter',
};
