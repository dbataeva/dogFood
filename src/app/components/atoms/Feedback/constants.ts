export const SNACK_BAR_VISIBILITY_TIME = 3000;
