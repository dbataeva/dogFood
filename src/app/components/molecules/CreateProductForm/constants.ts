import { number, object, string } from 'yup';

export const TEXT_MAP = {
	name: 'Наименование:',
	price: 'Цена:',
	discount: 'Скидка:',
	pictures: 'Ссылка на изображение:',
	stock: 'Количество в наличии:',
	description: 'Описание товара:',
	finish: 'Добавить продукт',
	successfulAdding: 'Продукт успешно добавлен',
	nameError: 'Имя должно быть заполнено',
	priceError: 'Цена должна быть заполнена',
	discountError: 'Скидка должна быть заполнена',
	picturesError: 'Добавьте ссылку на изобржения продукта',
	stockError: 'Количество в наличии должно быть заполнено',
	descriptionError: 'Добавьте описание продукта',
};

export const createProductScheme = object({
	name: string().required(),
	price: number().required(),
	discount: number().required(),
	pictures: string().url().required(),
	stock: number().required(),
	description: string().required(),
});

export const TEST_ID_MAP = {
	form: 'form',
	nameField: 'nameField',
	priceField: 'priceField',
	discountField: 'discountField',
	picturesField: 'picturesField',
	stockField: 'stockField',
	descriptionField: 'descriptionField',
	submitButton: 'submitButton',
};
