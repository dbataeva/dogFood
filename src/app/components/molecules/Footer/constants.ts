export const TEXT_MAP = {
	catalog: 'Каталог',
	sale: 'Акции',
	news: 'Новости',
	review: 'Отзывы',
	paymentAndDelivery: 'Оплата и доставка',
	fac: 'Часто спрашивают',
	feedback: 'Обратная связь',
	contacts: 'Контакты',
};

export const LEFT_MENU_ITEMS = [
	TEXT_MAP.catalog,
	TEXT_MAP.sale,
	TEXT_MAP.news,
	TEXT_MAP.review,
];

export const RIGHT_MENU_ITEMS = [
	TEXT_MAP.paymentAndDelivery,
	TEXT_MAP.fac,
	TEXT_MAP.feedback,
	TEXT_MAP.contacts,
];

export const TEST_ID_MAP = {
	footer: 'footer',
	logo: 'footerLogo',
	contacts: 'contacts',
};
