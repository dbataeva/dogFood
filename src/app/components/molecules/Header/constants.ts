export const MOBILE_MENU_ID = 'primary-search-account-menu-mobile';

export const HEADER_TEST_ID_MAP = {
	header: 'header',
	logo: 'headerLogo',
	searchBar: 'searchBar',
	navIcons: 'navIcons',
};
