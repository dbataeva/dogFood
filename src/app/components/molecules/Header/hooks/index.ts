export { useSearch } from './useSearch';
export { useIconButtons } from './useIconButtons';
