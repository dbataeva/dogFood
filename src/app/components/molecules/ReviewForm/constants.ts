import { object, number, string } from 'yup';

const MIN_RATING = 1;
const MAX_RATING = 5;

export const reviewScheme = object({
	rating: number().min(MIN_RATING).max(MAX_RATING).required(),
	text: string().required(),
});

export const TEXT_MAP = {
	addReview: 'Добавьте свой отзыв:',
	rating: 'Оценка',
	cancel: 'Отменить',
	save: 'Сохранить',
	successAddReview: 'Отзыв успешно добавлен!',
	ratingError: 'Рейтинг долже быть числом от 1 до 5',
	textError: 'Отзыв должен состоять как минимум из одного симовла',
	text: 'Отзыв',
};

export const TEST_ID_MAP = {
	form: 'form',
	ratingField: 'ratingField',
	reviewField: 'reviewField',
	submitButton: 'submitButton',
};
