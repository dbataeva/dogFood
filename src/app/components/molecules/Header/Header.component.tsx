import { FC, memo, useState } from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';

import { Logo, MobileMenu, Search, HeaderIconButtons } from '../../atoms';
import { MOBILE_MENU_ID, TEST_ID_MAP } from './constants';
import { useIconButtons, useSearch } from './hooks';

export const Header: FC = memo(() => {
	const [mobileMoreAnchorEl, setMobileMoreAnchorEl] =
		useState<null | HTMLElement>(null);

	const handleMobileMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
		setMobileMoreAnchorEl(event.currentTarget);
	};
	const handleMobileMenuClose = () => {
		setMobileMoreAnchorEl(null);
	};

	const showSearch = useSearch();
	const {
		showIconButtons,
		exitIconClickHandler,
		basketIconClickHandler,
		profileIconClickHandler,
		favoriteIconClickHandler,
	} = useIconButtons(handleMobileMenuClose);

	return (
		<Box
			data-testid={TEST_ID_MAP.header}
			sx={{ flexGrow: 1, marginBottom: '20px' }}>
			<AppBar position={'static'}>
				<Toolbar sx={{ display: 'flex', justifyContent: 'space-between' }}>
					<IconButton
						size={'large'}
						edge={'start'}
						color={'inherit'}
						aria-label={'open drawer'}
						sx={{ mr: 2 }}>
						<Logo testId={TEST_ID_MAP.logo} />
					</IconButton>
					{showSearch && <Search testId={TEST_ID_MAP.searchBar} />}
					{showIconButtons && (
						<HeaderIconButtons
							testId={TEST_ID_MAP.navIcons}
							mobileMenuId={MOBILE_MENU_ID}
							exitIconClickHandler={exitIconClickHandler}
							handleMobileMenuOpen={handleMobileMenuOpen}
							basketIconClickHandler={basketIconClickHandler}
							profileIconClickHandler={profileIconClickHandler}
							favoriteIconClickHandler={favoriteIconClickHandler}
						/>
					)}
				</Toolbar>
			</AppBar>
			{showIconButtons && (
				<MobileMenu
					mobileMenuId={MOBILE_MENU_ID}
					mobileMoreAnchorEl={mobileMoreAnchorEl}
					exitIconClickHandler={exitIconClickHandler}
					handleMobileMenuClose={handleMobileMenuClose}
					basketIconClickHandler={basketIconClickHandler}
					profileIconClickHandler={profileIconClickHandler}
					favoriteIconClickHandler={favoriteIconClickHandler}
				/>
			)}
		</Box>
	);
});

Header.displayName = 'Header';
