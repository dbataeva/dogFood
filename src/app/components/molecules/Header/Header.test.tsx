import { screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { userEvent } from '@testing-library/user-event';
import * as router from 'react-router';
import ReactTestRenderer from 'react-test-renderer';

import { HEADER_TEST_ID_MAP } from './constants';
import { testProvidersFn } from '../../../testProviders';
import { Header } from './Header.component';

describe('Компонент Header', () => {
	it('должен рендериться', () => {
		testProvidersFn();
		expect(screen.getByTestId(HEADER_TEST_ID_MAP.header)).toBeInTheDocument();
	});

	describe('компоненты внутри Header', () => {
		describe('Logo', () => {
			it('должно рендериться', () => {
				testProvidersFn();
				expect(screen.getByTestId(HEADER_TEST_ID_MAP.logo)).toBeInTheDocument();
			});

			it('должен быть переход на главную страницу при клике на лого', () => {
				const navigate = jest.fn();

				jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
				testProvidersFn('/basket', true);

				const logo = screen.getByTestId(HEADER_TEST_ID_MAP.logo);

				userEvent.click(logo);

				waitFor(() => expect(navigate).toHaveBeenCalledWith('/'));
			});
		});

		describe('строка поиска', () => {
			it('должна рендериться на главной странице', () => {
				testProvidersFn('/', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.searchBar)
				).toBeInTheDocument();
			});

			it('должна отсутствовать на странице регистрации', () => {
				testProvidersFn('/signUp');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице входа', () => {
				testProvidersFn('/signIn');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице детальной карточки продукта', () => {
				testProvidersFn('/products/123', true);
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице избранных продуктво', () => {
				testProvidersFn('/products/favorite', true);
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице детальной карточки пользователя', () => {
				testProvidersFn('/users/me', true);
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице корзины', () => {
				testProvidersFn('/basket', true);
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});

			it('должна отсутствовать на странице ошибки', () => {
				testProvidersFn('/qqq');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.searchBar)
				).not.toBeInTheDocument();
			});
		});

		describe('иконки навигации', () => {
			it('должны рендериться на главной странице', () => {
				testProvidersFn('/', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.navIcons)
				).toBeInTheDocument();
			});

			it('должны отсутствовать на странице регистрации', () => {
				testProvidersFn('/signUp');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.navIcons)
				).not.toBeInTheDocument();
			});

			it('должны отсутствовать на странице входа', () => {
				testProvidersFn('/signIn');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.navIcons)
				).not.toBeInTheDocument();
			});

			it('должны рендериться на странице детальной карточки продукта', () => {
				testProvidersFn('/products/123', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.navIcons)
				).toBeInTheDocument();
			});

			it('должны рендериться на странице избранных продуктво', () => {
				testProvidersFn('/products/favorite', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.navIcons)
				).toBeInTheDocument();
			});

			it('должны рендериться на странице детальной карточки пользователя', () => {
				testProvidersFn('/users/me', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.navIcons)
				).toBeInTheDocument();
			});

			it('должны рендериться на странице корзины', () => {
				testProvidersFn('/basket', true);
				expect(
					screen.getByTestId(HEADER_TEST_ID_MAP.navIcons)
				).toBeInTheDocument();
			});

			it('должны отсутствовать на странице ошибки', () => {
				testProvidersFn('/qqq');
				expect(
					screen.queryByTestId(HEADER_TEST_ID_MAP.navIcons)
				).not.toBeInTheDocument();
			});
		});
	});

	describe("должен соответствовать snapshot'ам", () => {
		beforeEach(() => {
			const navigate = jest.fn();

			jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
		});

		it("должен соответствовать snapshot'у на странице авторизации", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'signUp',
				pathname: '/signUp',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на странице входа в приложение", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'signIn',
				pathname: '/signIn',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на главной странице", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'default',
				pathname: '/',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на детальной странице продукта", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'default',
				pathname: '/products/123',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на странице избранных продуктов", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'default',
				pathname: '/products/favorite',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на странице пользователя", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'default',
				pathname: '/users/me',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});

		it("должен соответствовать snapshot'у на странице корзины", () => {
			jest.spyOn(router, 'useLocation').mockReturnValue({
				hash: '',
				key: 'default',
				pathname: '/basket',
				search: '',
				state: null,
			});

			const tree = ReactTestRenderer.create(<Header />).toJSON();

			expect(tree).toMatchSnapshot();
		});
	});
});
