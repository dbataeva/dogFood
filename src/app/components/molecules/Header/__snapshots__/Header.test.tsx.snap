// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на главной странице 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на детальной странице продукта 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на странице авторизации 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на странице входа в приложение 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на странице избранных продуктов 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на странице корзины 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Компонент Header должен соответствовать snapshot'ам должен соответствовать snapshot'у на странице пользователя 1`] = `
<div
  className="MuiBox-root css-125p8oa"
  data-testid="header"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          data-testid="headerLogo"
          onClick={[Function]}
          style={
            {
              "color": "white",
            }
          }
        />
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
